src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaConsumerImpl.java

```diff
@@ -221,6 +227,14 @@ public void unsubscribe() {
 
   @Override
   public ConsumerRecords<K, V> poll(long timeout) {
+    if (_lastProcessedResult != null && _lastProcessedResult.exception() != null) {
```

Comment:
Shouldn't offsets be modified after the call to the consumer records processor returns?  That would have similar semantics to the existing autocommit.